### PR TITLE
conformance-profile: Remove experimental build tag

### PR DESCRIPTION
**Pull request by sayboras**
**What type of PR is this?**

/kind test
/area conformance

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Relates: #2242

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

Testing was done as part of https://github.com/cilium/cilium/pull/27596

### CODE REVIEW

File: conformance/apis/v1alpha1/doc.go [hunk 1]
```diff
@@ -23,17 +20,10 @@ limitations under the License.
...
-// features, then you should not use them. If you would like to opt into these
-// unreleased features use Go build tags to enable them, e.g.:
-//
-//   $ GOFLAGS='-tags=experimental' go test ./conformance/... -args ${CONFORMANCE_ARGS}
+//   $ go test ./conformance/... -args ${CONFORMANCE_ARGS}
 //
-// Please note that everything here is considered experimental and subject to
-// change. Expect breaking changes and/or complete removals if you start using
-// them.
+// While the experiment build tag is removed, please note that everything here
```
Comment (mlavacca) [inline conformance/apis/v1alpha1/doc.go:25]: I wonder if we really need to mention the experimental build tag here. From a user perspective, this preamble can be meaningless, as such a tag does not exist anymore.

**Review by mlavacca — COMMENTED** [review]
Thanks for this PR, @sayboras!

File: conformance/apis/v1alpha1/doc.go (hunk 1 above)
Comment (arkodg) [reply conformance/apis/v1alpha1/doc.go:25]: +1, vote for keeping as is

Comment (kflynn) [reply conformance/apis/v1alpha1/doc.go:25]: I'd probably also not mention the experimental build tag here.

**Review by kflynn — CHANGES_REQUESTED** [review]
...hmmmm. When I tried a conformance run from the CLI with the `experimental` build tag enabled, it tried to run `TestConformance` and then `TestExperimentalConformance`, and `TestExperimentalConformance` failed because it was expecting an empty cluster to play in. 

This PR leaves both functions intact, and I would expect it to fail because of that. If the idea is that the stuff previously tagged with `experimental` is now OK to include everywhere, I'd expect to see `TestExperimentalConformance` go away?

File: conformance/apis/v1alpha1/doc.go (hunk 1 above)
Comment (sayboras) [reply conformance/apis/v1alpha1/doc.go:25]: good point, let me revert this change.

**Comment by sayboras** [comment]
> I'd expect to see TestExperimentalConformance go away?

Yes, understand that cleaning up of similar/duplicated code is required to
have a clean conformance test framework, however, such work might delay
v1.0.0 GA release, hence, I only remove the build tag to remove the soft
blocker in this PR. Subsequent cleanup/optimization can be done later.

**Comment by sayboras** [comment]
The previous CI failure was due to recent renaming in main (e.g. HTTPExtendedFeatures -> HTTPRouteExtendedFeatures). The current failure seems to be unrelated to this PR.

**Comment by mlavacca** [comment]
@sayboras, the linter is complaining. It looks like you need to do some `go` and `yaml` linting.

**Review by shaneutt — APPROVED** [review]
/approve
/lgtm

**Review by arkodg — APPROVED** [review]
/approve
/lgtm

**Pull request merged**